core/src/mender-client.c

```diff
@@ -244,13 +251,17 @@ mender_err_t
 mender_client_init(mender_client_config_t *config, mender_client_callbacks_t *callbacks) {
 
     assert(NULL != config);
-    assert(NULL != config->device_type);
     assert(NULL != callbacks);
     assert(NULL != callbacks->restart);
     mender_err_t ret;
 
-    mender_client_config.device_type = config->device_type;
-
+    if ((NULL != config->device_type) && (0 != strlen(config->device_type))) {
```

Comment:
It is a good call to leave the device type, which aligns the API with the server url, for example

Reply:
Leave the device type?

Reply:
I was referring to his point in the PR description:

> Ticket [MEN-7547](https://northerntech.atlassian.net/browse/MEN-7547) says to remove device_type from mender_client_config_t. But I don't see the reason why we cannot have both.